# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/ia2/ia2AccessibleRelation.h
# Commit: 8d31c0604efe
# Full Hash: 8d31c0604efef2015995df32766d38dd4dd45ca2
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-08-04 04:34:13
# Regressor Bug: 1781536
# File Overlap Count: 1
# Description:
#   Bug 1781536 part 1: Support base Accessible in Relation. r=morgan
#   
#   Previously, Relation::Next returned a LocalAccessible.
#   Now, it returns an Accessible so that it can support RemoteAccessible as well.
#   Relation::LocalNext has been added as a convenience to return a LocalAccessible when the caller is certain that only LocalAccessible is relevant.
# ==============================================================================

diff -r af7f9589f48c -r 8d31c0604efe accessible/windows/ia2/ia2AccessibleRelation.h
--- a/accessible/windows/ia2/ia2AccessibleRelation.h	Thu Aug 04 02:47:34 2022 +0300
+++ b/accessible/windows/ia2/ia2AccessibleRelation.h	Wed Aug 03 23:58:50 2022 +0000
@@ -8,7 +8,7 @@
 #ifndef _NS_ACCESSIBLE_RELATION_WRAP_H
 #define _NS_ACCESSIBLE_RELATION_WRAP_H
 
-#include "LocalAccessible.h"
+#include "MsaaAccessible.h"
 #include "IUnknownImpl.h"
 
 #include <utility>
@@ -23,10 +23,6 @@
  public:
   ia2AccessibleRelation(RelationType aType, Relation* aRel);
 
-  ia2AccessibleRelation(RelationType aType,
-                        nsTArray<RefPtr<LocalAccessible>>&& aTargets)
-      : mType(aType), mTargets(std::move(aTargets)) {}
-
   // IUnknown
   DECL_IUNKNOWN
 
@@ -57,7 +53,7 @@
   ia2AccessibleRelation& operator=(const ia2AccessibleRelation&);
 
   RelationType mType;
-  nsTArray<RefPtr<LocalAccessible>> mTargets;
+  nsTArray<RefPtr<MsaaAccessible>> mTargets;
 };
 
 /**